### ISSUE TITLE
ActivityPub: Send out Delete activities for users and serve a proper response when the deleted user object is requested

### ISSUE THREAD

**Issue by rinpatch** [friendica/friendica#6943]
### Expected behavior
When the user is deleted the server would send out Delete activities.

When the user is requested with `Accept: application/activity+json` header the server would serve [either a Tombstone object with a 410 Gone status code, or a 404 Not Found with an empty response body](https://www.w3.org/TR/activitypub/#delete-activity-outbox)
### Actual behavior
When the user is deleted the server does not appear to send out Delete activities

When the user is requested with `Accept: application/activity+json` header it returns  an empty array with 200 OK status code, which is not a valid AS2 object, so it confuses other fedi software
### Steps to reproduce the problem
1. Delete a user account
### Friendica version you encountered the problem

Tried on https://tryfriendica.de and https://nerdica.net

### Friendica source (git, zip)
N/A
### PHP version
N/A
### SQL version
N/A

**Comment by MrPetovan** [comment]
This is a duplicate of #6553, it was fixed in https://github.com/friendica/friendica/commit/0615c8438abf3519156d2ba9c0bf378c08f44737 and merged in https://github.com/friendica/friendica/pull/6562 for the 2019.03 release.

https://github.com/friendica/friendica/blob/854cc3e47296d4a1b01b893376064cf254d84f79/src/Module/Profile.php#L60-L63

Can anyone else confirms this? Can you please share a deleted username on either of those nodes?

**Comment by rinpatch** [comment]
@MrPetovan https://tryfriendica.de/profile/oof and https://nerdica.net/profile/skin

sorry in advance if it is really a duplicate

**Comment by tobiasd** [comment]
Both mentioned nodes are running 2019.03. So assuming @rinpatch did recent tests with it, #6562 does not work.

**Comment by rinpatch** [comment]
yeah, I tested it just today, you can see it for yourself by executing `curl -H "Accept: application/activity+json" https://nerdica.net/profile/skin`

**Comment by MrPetovan** [comment]
> sorry in advance if it is really a duplicate

It's okay, if it happening again it needs to be addressed again. I was just providing historical context into this issue we already had reported and thought we had fixed.

**Comment by annando** [comment]
The delete message to the receivers is sent with a low priority. On bigger servers this can last minutes or even hours until this is processed.

**Comment by lanodan** [comment]
And for the array received when fetching a profile?

**Comment by MrPetovan** [comment]
I tested on my personal node and hitting a deleted profile URL with the json header correctly returned the 410 error. We would need to get in touch with either admin to determine the actual status of both these profiles.

**Comment by rinpatch** [comment]
Huh, weird. Also [was noticed on libranet.de](https://git.pleroma.social/pleroma/pleroma/issues/754#note_22416) if that helps

**Comment by MrPetovan** [comment]
Confirmed.

I looked into it, and we are returning the 410 status code in a very specific scenario: there isn't a `user` record anymore, and there's a `userd` (list of usernames that can't be used anymore) record.

Unfortunately, ìf there's a user record with the following flags set, it will output a 200 status code with no data:
- `blocked` set;
- OR `account_expired` set;
- OR `account_removed` set;
- OR `verified` unset.

These fields corresponds to various moments of the user account life cycle and if they shouldn't all result in a 410 status code, they definitely shouldn't result in a 200 code.

I'll submit a patch for this.
